lib/storage/index_db_test.go

```diff
@@ -2216,3 +2216,194 @@ func stopTestStorage(s *Storage) {
 	s.tsidCache.Stop()
 	fs.MustRemoveDirAtomic(s.cachePath)
 }
+
+func TestSearchContainsTimeRange(t *testing.T) {
+	const path = "TestSearchContainsTimeRange"
+	os.RemoveAll(path)
+	s := MustOpenStorage(path, retentionMax, 0, 0)
+	db := s.idb()
+
+	is := db.getIndexSearch(0, 0, noDeadline)
+
+	// Create a bunch of per-day time series
+	const days = 6
+	const tenant2IngestionDay = 8
+	const metricsPerDay = 1000
+	theDay := time.Date(2019, time.October, 15, 5, 1, 0, 0, time.UTC)
+	now := uint64(timestampFromTime(theDay))
+	baseDate := now / msecPerDay
```

Comment:
I'd rename theDay, now, and baseDate to rotationDay, rotationMillis, and rotationDate so it is clear that 1) they are representations of the same point in time and 2) and point in time is when we rotate the indexdb.